torat_server/client_ishell.go

```diff
@@ -257,10 +262,26 @@ func (client *activeClient) Shred(c *ishell.Context) {
 	}
 }
 
-// ping something?
-// TODO: I don't think this feature works
-func (client *activeClient) Ping(c *ishell.Context) error {
-	return client.RPC.Call("API.Ping", void, &void)
+// Speedtest the clients internet connection
+func (client *activeClient) Speedtest(c *ishell.Context) {
+	c.ProgressBar().Indeterminate(true)
+	c.ProgressBar().Start()
+
+	r := shared.Speedtest{}
+	if err := client.RPC.Call("API.Speedtest", void, &r); err != nil {
+		c.ProgressBar().Final(yellow("["+client.Client.Name+"] ") + red("[!] Could not perform speedtest on client!"))
+		c.ProgressBar().Final(yellow("["+client.Client.Name+"] ") + red("[!] ", err))
```

Comment:
c.Progressbar().Final should only be called once

Please print the error in the first call, e.g. `red("Could not perform speedtest:", err)`

Reply:
whoops, thats a bad. Not sure why my editor didn't catch that